File: src/Shared/Dtos/FileStatisticsDto.cs
namespace Shared.Dtos;

/// <summary>
/// Statistics about indexed files and duplicates.
/// </summary>
public record FileStatisticsDto
{
    public int TotalFiles { get; init; }
    public long TotalSizeBytes { get; init; }
    public int DuplicateGroups { get; init; }
    public int DuplicateFiles { get; init; }
    public long PotentialSavingsBytes { get; init; }
    public DateTime? LastIndexedAt { get; init; }
}

File: src/Web/src/app/models/file-statistics.dto.ts
/**
 * Statistics about indexed files and duplicates.
 * Matches backend: src/Shared/Dtos/FileStatisticsDto.cs
 */
export interface FileStatisticsDto {
  totalFiles: number;
  totalSizeBytes: number;
  duplicateGroups: number;
  duplicateFiles: number;
  potentialSavingsBytes: number;
  lastIndexedAt: string | null;
}
